client/ext/bigquery/current-timestamp-optimizer_test.js

```diff
@@ -0,0 +1,264 @@
+/**
+ * @copyright Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @fileoverview Tests for SidebarTabService, holds state and data for the
+ * explorer sidebar.
+ *
+ * @author joemu@google.com (Joe Allan Muharsky)
+ */
+'use strict'
+
+goog.require('p3rf.perfkit.explorer.ext.bigquery.CurrentTimestampOptimizer');
+goog.require('p3rf.perfkit.explorer.ext.BigQuery.CurrentTimestampGranularity');
+
+
+goog.scope(function() {
+  const explorer = p3rf.perfkit.explorer;
+  const CurrentTimestampOptimizer = explorer.ext.bigquery.CurrentTimestampOptimizer;
+  const CurrentTimestampGranularity = explorer.ext.bigquery.CurrentTimestampGranularity;
+
+  describe('CurrentTimestampOptimizer', function() {
+    let testOptimizer, dashboard, dashboardSetting, widget, widgetSetting;
+    let PROVIDED_DASHBOARD, PROVIDED_WIDGET, PROVIDED_DATE;
+
+    const PROVIDED_YEAR = 2006;
+    const PROVIDED_MONTH = 1;
+    const PROVIDED_DAY = 15;
+    const PROVIDED_HOUR = 8;
+    const PROVIDED_MINUTE = 14;
+    const PROVIDED_SECOND = 23;
+
+    beforeEach(function() {
+      testOptimizer = new CurrentTimestampOptimizer();
+
+      PROVIDED_DATE = new Date(
```

Comment:
Do any of the tests still depend on this PROVIDED_DATE? If not, please remove it.

If it is still needed, I think it would be cleaner to inline the values here (looks like you're not using the constants anywhere else) and use a Date.UTC call to make it timezone independent.